AsyncDisplayKit/Private/ASInternalHelpers.m

```diff
@@ -84,6 +84,48 @@ void ASPerformBackgroundDeallocation(id object)
   [[ASDeallocQueue sharedDeallocationQueue] releaseObjectInBackground:object];
 }
 
+BOOL ASClassRequiresMainThreadDeallocation(Class class)
+{
+  if (class == [UIImage class] || class == [UIColor class]) {
+    return NO;
+  }
+
+  if ([class isSubclassOfClass:[UIResponder class]]
+      || [class isSubclassOfClass:[CALayer class]]
+      || [class isSubclassOfClass:[UIGestureRecognizer class]]) {
+    return YES;
+  }
+
+  const char *name = class_getName(class);
+  if (strncmp(name, "UI", 2) == 0 || strncmp(name, "AV", 2) == 0 || strncmp(name, "CA", 2) == 0) {
```

Comment:
If this name-based check is faster than the three `isSubclassOfClass` tests above, consider performing this check first.